sql/plan/alter_table.go

```diff
@@ -53,6 +53,127 @@ func (r *RenameTable) String() string {
 	return fmt.Sprintf("Rename table %s to %s", r.OldNames, r.NewNames)
 }
 
+func (r *RenameTable) RowIter(ctx *sql.Context, row sql.Row) (sql.RowIter, error) {
+	// TODO: 'ALTER TABLE newViewName RENAME view1' should fail on renaming views - should be fixed in vitess
+	renamer, ok := r.Db.(sql.TableRenamer)
+	if !ok {
+		return nil, sql.ErrRenameTableNotSupported.New(r.Db.Name())
+	}
+	viewDb, vok := r.Db.(sql.ViewDatabase)
+
+	for i, oldName := range r.OldNames {
+		if success, err := r.attemptRenameTable(ctx, oldName, r.NewNames[i]); err != nil {
+			return nil, err
+		} else if success {
+			err = renamer.RenameTable(ctx, oldName, r.NewNames[i])
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			if success, err = r.attemptRenameView(ctx, viewDb, vok, oldName, r.NewNames[i]); err != nil {
+				return nil, err
+			} else if !success {
+				return nil, sql.ErrTableNotFound.New(oldName)
+			}
+		}
+	}
+
+	return sql.RowsToRowIter(sql.NewRow(types.NewOkResult(0))), nil
+}
+
+func (r *RenameTable) attemptRenameTable(ctx *sql.Context, oldName, newName string) (bool, error) {
```

Comment:
I think tweaking the naming and responsibilities of these two functions (attemptRenameTable and attemptRenameView) would help make this code much easier to understand... 

I don't think you need the "attempt" prefix – these functions don't seem different from any other functions that could fail and return an error, so just calling it "renameView" would be a little more consistent. 

The "attempRenameTable" function is also pretty misleading... it won't actually try to rename the table, but it will update foreign keys, and then the original caller has to remember to call another function to actually rename the table. This makes the code a little harder to understand. The ideal interface seems like it would be something like "renameTable" and it would take care of everything for the caller, instead of leaving it in a state where the foreign keys are updated, but the table hasn't been renamed yet.